src/univers/version_range.py

```diff
@@ -868,6 +868,39 @@ def from_native(cls, string):
         return cls(constraints=constraints)
 
 
+class OpensslVersionRange(VersionRange):
+    """
+    Openssl version range.
+    openssl doesn't use <,>,<= or >=
+    For more see 'https://www.openssl.org/news/vulnerabilities.xml'
+
+    For exmaple::
+    >>> from univers.versions import OpensslVersion
+    >>> constraints = (
+    ... VersionConstraint(version=OpensslVersion("1.0.1af")),
+    ... VersionConstraint(comparator="=", version=OpensslVersion("3.0.1")),
+    ... VersionConstraint(comparator="=", version=OpensslVersion("1.1.1nf")),
+    ... )
+    >>> range = OpensslVersionRange(constraints=constraints)
+    >>> assert str(range) == 'vers:openssl/1.0.1af|1.1.1nf|3.0.1'
+    """
+
+    scheme = "openssl"
+    version_class = versions.OpensslVersion
+    vers_by_native_comparators = {"=": "="}
+
+    @classmethod
+    def from_native(cls, string):
+        cleaned = remove_spaces(string).lower()
+        constraints = set()
```

Comment:
Why a set and not a list?

Reply:
Just to avoid having possible duplicates

Reply:
why would there be possible duplicates? There are no other VersionRange that uses sets this way for constraints.
It is likely a good idea to avoid duplicated constraints (and I should add this to the vers spec)  but it should not be the responsability of a subclass... rather it would be done in one single place here https://github.com/nexB/univers/blob/6f03cf67d094fd823aca72bbb5352ccfcb0936fe/src/univers/version_range.py#L53 and properly tested. Let's do this in another PR... can you enter an issue for this?

Reply:
Thanks, then i don't see any chance for possible duplication. Will create an issue for adding some tests there.